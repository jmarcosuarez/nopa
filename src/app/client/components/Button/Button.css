@import "../../../static/stylesheets/abstracts/all.css";

:root {
  --spacer: 1rem;
}

.buttonWrapper {
  display: flex;
    justify-content: center;
}

.button {
  display: block;
  width: 100%;
  border-radius: 2px;
  font-family: 'Montserrat', sans-serif;
  font-size: 14px;
  font-weight: normal;
  background: var(--color-orange);
  margin: var(--spacer);
  padding: var(--spacer)  calc(var(--spacer) * 1.5);
  color: #FFFFFF;
  text-decoration: none;
  text-align: center;
  display: block;
  max-width: 20rem;
  border: 0;
  cursor: pointer;
  border: 1px solid transparent;
  box-shadow: 0 15px 35px rgba(50,50,93,.1),0 5px 15px rgba(0,0,0,.07);

  &:focus,
  &:hover {
  background-color: color(var(--color-orange) a(90%)); /* As rgba(255, 0, 0, .9)*/
  box-shadow: 0 18px 35px rgba(50,50,93,.1),0 8px 15px rgba(0,0,0,.07);
  }

  &:focus {
  outline: 0;
  }
}