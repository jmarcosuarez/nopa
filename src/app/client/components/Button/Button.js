import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import styles from './Button.css';

const Button = props => (
  <div className={styles.buttonWrapper}>
    <Link {...props} className={`btn ${styles.button}`}>
      {props.children}
    </Link>
  </div>
  );

Button.propTypes = {
  children: PropTypes.string.isRequired,
};

export default Button;
