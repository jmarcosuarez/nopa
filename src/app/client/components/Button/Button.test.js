/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './Button';

test('Renders correctly', () => {
  const wrapper = shallow(
    <Button>Send</Button>,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});