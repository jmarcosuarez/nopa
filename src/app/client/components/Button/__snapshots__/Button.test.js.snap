// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="buttonWrapper"
>
  <Link
    className="btn button"
    onlyActiveOnIndex={false}
    style={Object {}}
  >
    Send
  </Link>
</div>
`;
